barretenberg/cpp/src/barretenberg/api/api_ultra_honk.hpp

```diff
@@ -0,0 +1,311 @@
+            auto ipa_proof = HonkProof(proof.begin() + honk_proof_with_pub_inputs_length, proof.end());
+            auto tube_honk_proof = HonkProof(proof.begin(), proof.end() + honk_proof_with_pub_inputs_length);
+            verified = verifier.verify_proof(proof, ipa_proof);
+        } else {
+            verified = verifier.verify_proof(proof);
+        }
+
+        return verified;
+    }
+
+  public:
+    bool check([[maybe_unused]] const Flags& flags,
+               [[maybe_unused]] const std::filesystem::path& bytecode_path,
+               [[maybe_unused]] const std::filesystem::path& witness_path) override
+    {
+        throw_or_abort("API function check_witness not implemented");
+        return false;
+    }
+
+    void prove(const API::Flags& flags,
+               const std::filesystem::path& bytecode_path,
+               const std::filesystem::path& witness_path,
+               const std::filesystem::path& output_dir) override
+    {
+        const auto _write = [&](auto&& _prove_output) {
+            write(_prove_output, flags.output_data_type, flags.output_content_type, output_dir);
+        };
+
+        const bool init = flags.init_kzg_accumulator;
+        const bool compute_vk = flags.output_content_type == "proof_and_vk";
+
+        if (flags.ipa_accumulation) {
+            _write(_prove<UltraRollupFlavor>(compute_vk, init, bytecode_path, witness_path));
+        } else if (flags.oracle_hash_type == "poseidon2") {
+            _write(_prove<UltraFlavor>(compute_vk, init, bytecode_path, witness_path));
+        } else if (flags.oracle_hash_type == "keccak" && !flags.zk) {
+            _write(_prove<UltraKeccakFlavor>(compute_vk, init, bytecode_path, witness_path));
+        } else if (flags.oracle_hash_type == "keccak" && flags.zk) {
+            _write(_prove<UltraKeccakZKFlavor>(compute_vk, init, bytecode_path, witness_path));
+        } else {
+            throw_or_abort("Invalid proving options specified in _prove");
+        };
+    };
+
+    /**
+     * @brief Verifies a client ivc proof and writes the result to stdout
+     *
+     * Communication:
+     * - proc_exit: A boolean value is returned indicating whether the proof is valid.
+     *   an exit code of 0 will be returned for success and 1 for failure.
+     *
+     * @param proof_path Path to the file containing the serialized proof
+     * @param vk_path Path to the serialized verification key of the final (MegaHonk) circuit in the stack
+     * @param accumualtor_path Path to the file containing the serialized protogalaxy accumulator
+     * @return true (resp., false) if the proof is valid (resp., invalid).
+     */
+    bool verify(const API::Flags& flags,
+                const std::filesystem::path& proof_path,
+                const std::filesystem::path& vk_path) override
+    {
+        const bool ipa_accumulation = flags.ipa_accumulation;
+        if (ipa_accumulation) {
```

Comment:
not sure there is already a warning if multiple of these are set

Reply:
Right these should be made exclusive of each other